microcosm_elasticsearch/store.py

```diff
@@ -0,0 +1,218 @@
+"""
+Abstraction layer for persistence operations.
+
+Allows upstream (e.g. HTTP) operations to use a consistent persistence interface
+and avoids leaking persistence abstractions throughout a code base.
+
+Intended to be duck-type compatible with `microcosm_postgres.store.Store`.
+
+"""
+from time import time
+from uuid import uuid4
+
+from microcosm_elasticsearch.errors import translate_elasticsearch_errors
+
+
+class Store(object):
+    """
+    Elasticsearch persistence interface.
+
+    """
+    def __init__(self, graph, index, doc_type):
+        """
+        :param graph: the object graph
+        :param index: the name of an index to use
+        :param doc_type: a `elasticsearch_dsl.DocType` subclass to persist.
```

Comment:
I'm using `elasticsearch_dsl` as a pseudo ORM. Most of the "interesting" work I did in my first draft (API calls, query abstraction) are already part of this DSL. It's not perfect, but it's certainly good enough.